src/records/RecCore.cc

```diff
@@ -560,18 +560,20 @@ RecLookupMatchingRecords(unsigned rec_type, const char *match, void (*callback)(
     return REC_ERR_FAIL;
   }
 
-  // First find the new metrics, this is a bit of a hack, beacuse we still use the old
-  // librecords callback with a "pseudo" record.
-  RecRecord tmp;
-
-  tmp.rec_type  = RECT_ALL;
-  tmp.data_type = RECD_INT;
-
-  for (auto &&[name, val] : ts::Metrics::instance()) {
-    if (regex.match(name.data()) >= 0) {
-      tmp.name         = name.data();
-      tmp.data.rec_int = val;
-      callback(&tmp, data);
+  if ((rec_type & (RECT_PROCESS | RECT_NODE | RECT_PLUGIN))) {
+    // First find the new metrics, this is a bit of a hack, because we still use the old
+    // librecords callback with a "pseudo" record.
+    RecRecord tmp;
+
+    tmp.rec_type  = RECT_PROCESS;
+    tmp.data_type = RECD_INT;
+
+    for (auto &&[name, val] : ts::Metrics::instance()) {
+      if (regex.match(name.data()) >= 0) {
+        tmp.name         = name.data();
+        tmp.data.rec_int = val;
+        callback(&tmp, data);
+      }
     }
   }
```

Comment:
all metrics will be handled here, so maybe we can just return from here and not lookup on the `g_records` which seems to hold only configs.

Reply:
Sounds reasonable for metrics lookup.